packages/react-ui-core/src/Gmap/FreeDrawLayer.js

```diff
@@ -0,0 +1,152 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import autobind from 'autobind-decorator'
+import themed from 'react-themed'
+import { EVENTS } from './utils/const'
+import { setupEvents, removeEvent } from './utils/mapEventHelpers'
+
+@themed(['Gmap_FreeDrawTootip'], { pure: true })
+export default class FreeDrawLayer extends PureComponent {
+  static propTypes = {
+    map: PropTypes.object,
+    onMapDrawStart: PropTypes.func,
+    onMapDrawEnd: PropTypes.func,
+    theme: PropTypes.object,
+    mapControls: PropTypes.object.isRequired,
+    shapes: PropTypes.Array,
+  }
+
+  static defaultProps = {
+    theme: {},
+    map: {},
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      polygons: [],
+    }
+  }
+
+  enableMapControls() {
+    const { map, mapControls } = this.props
+    map.setOptions({
+      draggable: true,
+      zoomControl: true,
+      scrollwheel: true,
+      disableDoubleClickZoom: true,
+      ...mapControls,
+    })
+  }
+
+  disableMapControls() {
```

Comment:
Just a heads up for when we implement this into ag.js, we will need to make sure to turn off our custom zoom event listeners as well, because we have searches firing on double clicks, scrollwheels etc independent of Google Map's zoom listeners. If we forget to turn them off, we'll potentially fire searches on those events even though the map is no longer listening for them.